api/db/postgres.go

```diff
@@ -0,0 +1,607 @@
+	}
+	return nil
+}
+
+// UpdateOneDBUser checks if a given user is present into user table and update it.
+func (pR *PostgresRequests) UpdateOneDBUser(
+	mapParams map[string]interface{}, updatedUser types.User) error {
+	if (types.User{}) == updatedUser {
+		return errors.New("Empty fields to be updated")
+	}
+	if len(mapParams) == 0 {
+		return errors.New("Empty fields to search")
+	}
+	updatedUserMap := map[string]interface{}{
+		"username":     updatedUser.Username,
+		"password":     updatedUser.Password,
+		"salt":         updatedUser.Salt,
+		"iterations":   updatedUser.Iterations,
+		"keylen":       updatedUser.KeyLen,
+		"hashfunction": updatedUser.HashFunction,
+	}
+	finalQuery, values := ConfigureUpdateQuery(
+		`UPDATE user`, mapParams, updatedUserMap)
+	rowsAff, err := pR.DataRetriever.WriteInDB(finalQuery, values)
+	if err != nil {
+		return err
+	}
+	if rowsAff == int64(0) {
+		return errors.New("No data was updated")
+	}
+	return nil
+}
+
+// UpdateOneDBAnalysisContainer checks if a given analysis is present into analysis table
+// and update the container associated in it.
+func (pR *PostgresRequests) UpdateOneDBAnalysisContainer(
+	mapParams, updateQuery map[string]interface{}) error {
+	if len(updateQuery) == 0 {
+		return errors.New("Empty fields to be updated")
+	}
+	if len(mapParams) == 0 {
+		return errors.New("Empty fields to search")
+	}
+	finalQuery, values := ConfigureUpdateQuery(
+		`UPDATE analysis`, mapParams, updateQuery)
+	rowsAff, err := pR.DataRetriever.WriteInDB(finalQuery, values)
+	if err != nil {
+		return err
+	}
+	if rowsAff == int64(0) {
+		return errors.New("No data was updated")
+	}
+	return nil
+}
+
+// UpdateOneDBAccessToken checks if a given access token is present into accessToken and update it.
+func (pR *PostgresRequests) UpdateOneDBAccessToken(
+	mapParams map[string]interface{}, updatedAccessToken types.DBToken) error {
+	if (types.DBToken{}) == updatedAccessToken {
+		return errors.New("Empty fields to be updated")
+	}
+	if len(mapParams) == 0 {
+		return errors.New("Empty fields to search")
+	}
+	updatedAccessTokenMap := map[string]interface{}{
+		"huskytoken":    updatedAccessToken.HuskyToken,
+		"repositoryURL": updatedAccessToken.URL,
+		"isValid":       updatedAccessToken.IsValid,
+		"createdAt":     updatedAccessToken.CreatedAt,
+		"salt":          updatedAccessToken.Salt,
+		"uuid":          updatedAccessToken.UUID,
+	}
+	finalQuery, values := ConfigureUpdateQuery(
+		`UPDATE accessToken`, mapParams, updatedAccessTokenMap)
+	rowsAff, err := pR.DataRetriever.WriteInDB(finalQuery, values)
+	if err != nil {
+		return err
+	}
+	if rowsAff == int64(0) {
+		return errors.New("No data was updated")
+	}
+	return nil
+}
+
+// GetMetricByType returns data about the metric received
+func (pR *PostgresRequests) GetMetricByType(
+	metricType string, queryStringParams map[string][]string) (interface{}, error) {
+	// TODO: Need to know how to generate the same statistics
```

Comment:
Hmm, would you mind creating an issue so that we can work on this later, please? It would be great to let people know that we basically need to convert all MongoDB queries from `huskystat.go` into SQL ones. I also prefer not to letting `TODO` comments in our code, not sure what do you think about it. 🙃